sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -441,10 +447,10 @@ private static void verifyUnsplittableMethods(ErrorReporter errors, DoFnSignatur
             DoFn.InputProvider.class.getSimpleName(),
             formatType(paramT),
             formatType(expectedInputProviderT));
-        extraParameters.add(DoFnSignature.Parameter.INPUT_PROVIDER);
+        extraParameters.add(DoFnSignature.Parameter.inputProvider());
```

Comment:
Given the use of "contains" everywhere, we should make it a documented requirement that the Parameter's implement a reasonable hash-code and equals?